Release notes + documentation on portaal taak object and custom front-end components (#462)
valtimo-platform/valtimo-documentation: using-valtimo/form-flow/create-custom-compotent.md

Reviewer: create-custom-component.md

# Creating a custom form flow component

> Sometimes it isn't possible to capture the requirements for a form in Form.io. In cases like this you can create a 

Reviewer: ```suggestion
Sometimes it is not possible to capture the requirements for a form in Form.io. In cases like this you can create a 
```

custom front-end component.

Any Angular component can be used as a step in a form flow. Handling the form needs to be done in the component
as well. To submit the data for the step, and go to the next step a `POST` call needs to be done to the

> `/api/v1/form-flow/{formFlowId}/step/{stepInstanceId}` endpoint. By providing a json body for this request, submission 

Reviewer: ```suggestion
`/api/v1/form-flow/{formFlowId}/step/{stepInstanceId}` endpoint. By providing a JSON body for this request, submission 
```

data can be stored for use in form flow. 

## Registering a custom component

The component needs to be registered in the angular application so that the form flow framework can use it. Registering
is done by adding a reference to the component to the providers array in the angular application. The 

> `FORM_FLOW_COMPONENT_TOKEN` token can be provided with an array of component registrations. Each one contains an `id` 

Reviewer: ```suggestion
`FORM_FLOW_COMPONENT_TOKEN` can be provided with an array of component registrations. Each one contains an `id` 
```
